moved/src/storage.rs

```diff
@@ -97,62 +102,48 @@ impl State for InMemoryState {
         Ok(())
     }
 
-    fn db(&self) -> &(impl TreeReader + Sync) {
-        &self.db
+    fn db(&self) -> Arc<impl DB> {
+        self.db.clone()
     }
 
     fn resolver(&self) -> &(impl MoveResolver<Self::Err> + TableResolver) {
         &self.resolver
     }
 
     fn state_root(&self) -> B256 {
-        self.version()
-            .map(|version| self.tree().get_root_hash(version).unwrap().to_h256())
-            .unwrap_or_default()
+        self.current_state_root.unwrap_or_default()
     }
 }
 
 impl InMemoryState {
-    /// This method return none only in case it is called before genesis i.e. state is completely
-    /// blank, there are no nodes in the trie.
-    fn version(&self) -> Option<Version> {
-        self.version.checked_sub(1)
-    }
-
-    fn next_version(&mut self) -> Version {
-        self.version += 1;
-        self.version - 1
-    }
-
     fn insert_change_set_into_merkle_trie(&mut self, change_set: &ChangeSet) -> B256 {
-        let version = self.next_version();
         let values = change_set.to_tree_values();
-        let value_set = values.into_iter().map(|(k, v)| {
-            (
-                k,
-                v.map(|x| bcs::to_bytes(&x).expect("Must serialize value")),
-            )
-        });
-        let (root_hash, tree_update_batch) = self
-            .tree()
-            .put_value_set(value_set, version)
-            .expect("Fails on duplicate key or storage read. In memory storage cannot fail.");
-
-        self.db
-            .write_tree_update_batch(tree_update_batch)
-            .expect("Fails on duplicate key or storage write. In memory storage cannot fail.");
 
-        root_hash.to_h256()
+        let mut trie = self.tree();
+        for (k, v) in values {
+            let key_bytes = k.key_hash();
+            let value_bytes = v
+                .as_ref()
+                .map(|x| bcs::to_bytes(x).expect("Value must serialize"));
```

Comment:
Is this the reason why the proofs don't come in as the Proposer expects?
As in, the tree keeps the `change_set` instead of an `account` info.

Reply:
Perhaps this is the reason. Depends on if the value hash is included in the proof or not. We'll have to look into it more.